Lib/test/test_cmd_line.py

```diff
@@ -118,13 +118,10 @@ def run_python(*args):
         self.assertEqual(out.rstrip(), b'{}')
         self.assertEqual(err, b'')
         # "-X showrefcount" shows the refcount, but only in debug builds
-        rc, out, err = run_python('-X', 'showrefcount', '-c', code)
+        rc, out, err = run_python('-I', '-X', 'showrefcount', '-c', code)
         self.assertEqual(out.rstrip(), b"{'showrefcount': True}")
         if Py_DEBUG:
-            # bpo-46417: Tolerate negative reference count which can occur
-            # because of bugs in C extensions. This test is only about checking
-            # the showrefcount feature.
-            self.assertRegex(err, br'^\[-?\d+ refs, \d+ blocks\]')
+            self.assertRegex(err, br'^\[\d+ refs, \d+ blocks\]')
```

Comment:
FYI, from my macOS

`````

test test_cmd_line failed -- Traceback (most recent call last):
  File "/Users/user/oss/cpython/Lib/test/test_cmd_line.py", line 124, in test_showrefcount
    self.assertRegex(err, br'^\[\d+ refs, \d+ blocks\]')
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: Regex didn't match: b'^\\[\\d+ refs, \\d+ blocks\\]' not found in b'[-1 refs, 0 blocks]\n'

`````

Reply:
Oh crap. Ok, let's tolerate negative ref count for now. Checking that refs <= 0 and blocks == 0 is already a step forward!

Reply:
cc @kumaraditya303

Reply:
I didn't account for negative refcount on other platforms as I only saw output on Linux where it is zero.

Reply:
For the record,
https://github.com/python/cpython/runs/5322493186?check_suite_focus=true
Ubuntu also failed with the same issue

`````
FAIL: test_showrefcount (test.test_cmd_line.CmdLineTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/cpython/cpython-ro-srcdir/Lib/test/test_cmd_line.py", line 124, in test_showrefcount
    self.assertRegex(err, br'^\[\d+ refs, \d+ blocks\]')
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: Regex didn't match: b'^\\[\\d+ refs, \\d+ blocks\\]' not found in b'[-1 refs, 0 blocks]\n'

`````